dagobah/core/job.py

```diff
@@ -439,8 +439,11 @@ def _serialize(self, include_run_logs=False, strict_json=False):
             result = json.loads(json.dumps(result, cls=StrictJSONEncoder))
         return result
 
-    def _implements_function(obj, functions):
+    def _implements_function(self, obj, functions):
         """ Checks for the existence of a method or list of methods """
+        if isinstance(functions, str):
+            functions = [functions]
```

Comment:
I figure to make this method general, I would accept either a string, or list of strings. This line converts the single string to a list with just one string, otherwise the `for expected_method in functions:` will loop through each char in the string.

There is no case currently that needs a list of functions (unless we want to get really explicit) so if you'd rather just simplify it to just have one string, I can change it to that.